### PR TITLE
Hugo's personal page

### PR DESCRIPTION
**Pull request by hdevelopergit**
## Description

Hugo's Personal page

screenshot personalpage

## Additional Information

- [x] I have read the [contributing docs](/scaffold-eth/scaffold-eth-2/blob/main/CONTRIBUTING.md) (if this is your first contribution)
- [x] This is not a duplicate of any [existing pull request](https://github.com/scaffold-eth/scaffold-eth-2/pulls)

## Related Issues

_Closes #{issue number}_

_Note: If your changes are small and straightforward, you may skip the creation of an issue beforehand and remove this section. However, for medium-to-large changes, it is recommended to have an open issue for discussion and approval prior to submitting a pull request._

Your ENS/address:

### CODE REVIEW

**Comment by derrekcoleman** [comment]
Hi @hdevelopergit! It's nice to meet you and learn about you from your personal page. Here are a few small suggestions to make it even cleaner 🦾 

- I recommend including a screenshot of your personal page (and other UI changes you make in the future). It's much faster for a reviewer to see your changes in the body of your PR message than to have to clone your branch, yarn install, yarn start, open the page, etc. A reviewer will usually do all of that anyway to check other things, but it's a good habit to get into.
   - Please add a screenshot of your personal page to the PR message. Update it there and/or in future comments if you make additional commits that change the UI.
- I love the way you use divs to separate your github, linkedin, and twitter. Clean look! I also appreciate the use of `max-w-xl` in your bio section. 
   - In my opinion, you could go further to style the top section so it feels more "contained" (just like the social "cards" you made at the bottom) instead of text floating in a big background. Totally up to you, just an idea :)
- Since you reuse the social media cards three times, you might consider refactoring the styling out as a single component that you invoke three times.
   - I think it's a good habit to practice, but again I'll leave it up to you if you're interested in the exercise.

I look forward to seeing what you buidl!

**Comment by hdevelopergit** [comment]
> Hi @hdevelopergit! It's nice to meet you and learn about you from your personal page. Here are a few small suggestions to make it even cleaner 🦾
> 
> * I recommend including a screenshot of your personal page (and other UI changes you make in the future). It's much faster for a reviewer to see your changes in the body of your PR message than to have to clone your branch, yarn install, yarn start, open the page, etc. A reviewer will usually do all of that anyway to check other things, but it's a good habit to get into.
>   
>   * Please add a screenshot of your personal page to the PR message. Update it there and/or in future comments if you make additional commits that change the UI.
> * I love the way you use divs to separate your github, linkedin, and twitter. Clean look! I also appreciate the use of `max-w-xl` in your bio section.
>   
>   * In my opinion, you could go further to style the top section so it feels more "contained" (just like the social "cards" you made at the bottom) instead of text floating in a big background. Totally up to you, just an idea :)
> * Since you reuse the social media cards three times, you might consider refactoring the styling out as a single component that you invoke three times.
>   
>   * I think it's a good habit to practice, but again I'll leave it up to you if you're interested in the exercise.
> * Was it intentional that you tag `#20` in your PR? Your description doesn't explain what your personal page has to do with Mayowa's, so maybe that was an accident.
>   
>   * If so, please delete `#20` from your PR message.
> 
> I look forward to seeing what you buidl!

Hi @derrekcoleman ! Nice too meet you. I will take into account what you say. To update the personal page I have to do a new pull request or I can update this one? Thanks!

**Comment by derrekcoleman** [comment]
@hdevelopergit all you have to do it push your next commit(s) to this same branch "hdevelopergit:2-PersonalPage". Those will automatically get added to this pull request and show up in this thread.

**Comment by hdevelopergit** [comment]
> Hello @hdevelopergit! Nice to meet you
> 
> I was going through your socials and noticed that the LinkedIn link is broken. It is fixed if you add `https://` at the beginning
> 
>  hugoHomePageBug.mp4

Thanks Wilman. I've modified.

**Comment by hdevelopergit** [comment]
> @hdevelopergit all you have to do it push your next commit(s) to this same branch "hdevelopergit:2-PersonalPage". Those will automatically get added to this pull request and show up in this thread.

Thanks @derrekcoleman I've done some modifications and pushed. Also I've added the image to the PR. Looking forward your approval. 😊

**Comment by derrekcoleman** [comment]
Thanks for adding the screenshot and trying out the refactor! Also shout out to @wildanvin for your well-documented support on this PR.

@carletex or @technophile-04, please merge when you get a chance (and/or give me write access to this repo 🙏)

**Review by derrekcoleman — APPROVED** [review]

File: packages/nextjs/app/builders/0x033ebeaBCC05eBaF5628b7bDDfacaDF1426efAA4/page.tsx
Comment (technophile-04) [inline packages/nextjs/app/builders/0x033ebeaBCC05eBaF5628b7bDDfacaDF1426efAA4/page.tsx:50]: Umm instead of inlining the array in jsx and using `DynamicHeroIcon` we could do something like: 

```tsx
import type { NextPage } from "next";
import { BellIcon, BriefcaseIcon, CodeBracketIcon } from "@heroicons/react/24/outline";

const myData = [
  ["Explore my projects on", "GitHub", "https://github.com/hdevelopergit", CodeBracketIcon],
  ["Connect with me on", "Linkedin", "https://www.linkedin.com/in/hugo-hernán-angarola-bb271a7a", BriefcaseIcon],
  ["Follow me on", "Twitter", "https://x.com/hugo33612165", BellIcon],
] as const;

const Hugo: NextPage = () => {
       return (
       {/* ... */}
        <nav className="flex justify-center items-center gap-8 flex-col sm:flex-row">
          {myData.map(([text, title, url, Icon]) => (
            <div
              key={title}
              className="flex flex-col bg-base-100 px-6 py-6 text-center items-center max-w-xs rounded-3xl"
            >
              {/* <BellIcon className="h-8 w-8 fill-secondary"/> */}
              <Icon className="h-8 w-8 fill-secondary" />
              <p>
                {text}{" "}
                <a href={url} target="_blank" className="link">
                  {title}
                </a>
              </p>
            </div>
          ))}
        </nav>
       {/* ... */}       
       )
 };
```

This way we don't need to use of hacky `DynamicHeroIcon` component and also easy for readability ?

File: packages/nextjs/app/builders/0x033ebeaBCC05eBaF5628b7bDDfacaDF1426efAA4/DynamicHeroIcon.tsx
Comment (technophile-04) [inline packages/nextjs/app/builders/0x033ebeaBCC05eBaF5628b7bDDfacaDF1426efAA4/DynamicHeroIcon.tsx:1]: Umm this seems a bit hacky solution and also not that worth since by doing `import * from '@heroicons/react/24/outline'` you are are importing every thing, maybe remove this checkout https://github.com/BuidlGuidl/batch6.buidlguidl.com/pull/21/files#r1639364692

File: packages/nextjs/app/builders/0x033ebeaBCC05eBaF5628b7bDDfacaDF1426efAA4/page.tsx
Comment (technophile-04) [inline packages/nextjs/app/builders/0x033ebeaBCC05eBaF5628b7bDDfacaDF1426efAA4/page.tsx:1]: Could you please also remove all the comments ?

**Comment by technophile-04** [comment]
Hey thanks @hdevelopergit, this is nice !! also thanks all for nice suggestions / discussion just added a couple of comments

File: packages/nextjs/app/builders/0x033ebeaBCC05eBaF5628b7bDDfacaDF1426efAA4/page.tsx
Comment (hdevelopergit) [reply packages/nextjs/app/builders/0x033ebeaBCC05eBaF5628b7bDDfacaDF1426efAA4/page.tsx:50]: I will review. Honestly I'm not a web developer so for me all of this is new :)
The solution I've impelemented I had found googling. I will try to approach. Thanks!

**Comment by derrekcoleman** [comment]
Thanks for the great feedback, @technophile-04!

@hdevelopergit, as someone who is "not a web developer," I admire the way you're taking all of our feedback as a chance to learn new things and level up your skills. Try out Shiv's ideas and let us know how it goes!

**Comment by hdevelopergit** [comment]
> Hey thanks @hdevelopergit, this is nice !! also thanks all for nice suggestions / discussion just added a couple of comments

thanks you 😊

**Comment by hdevelopergit** [comment]
> Thanks for the great feedback, @technophile-04!
> 
> @hdevelopergit, as someone who is "not a web developer," I admire the way you're taking all of our feedback as a chance to learn new things and level up your skills. Try out Shiv's ideas and let us know how it goes!

thanks @derrekcoleman for your words 😊 Well, I want to learn. And challenges are a funny way to learn! Altough I must admit I'm a bit anxius to get my personal page approved. 🥺🤣

File: packages/nextjs/app/builders/0x033ebeaBCC05eBaF5628b7bDDfacaDF1426efAA4/page.tsx
Comment (hdevelopergit) [reply packages/nextjs/app/builders/0x033ebeaBCC05eBaF5628b7bDDfacaDF1426efAA4/page.tsx]: Hi @technophile-04 . I've implemented your approach and works perfectly. It's much more elegant this way. Thanks for suggestion! I've also deleted the comments in the coding. I hope now for the final approval 🙏😊

**Review by technophile-04 — APPROVED** [review]
Nice and clean!! Thanks @hdevelopergit ! and also others for great suggestions !

**Comment by hdevelopergit** [comment]
> Nice and clean!! Thanks @hdevelopergit ! and also others for great suggestions !

Thanks @technophile-04 ! I'm happy with my first merge 🥳
Now I can delete the branch correct? The fork I guess should continue open for next branches.

Thanks also @derrekcoleman and @wildanvin for help and suggestions!

**Pull request merged**
